week3/homework/src/actions.js

```diff
@@ -0,0 +1,100 @@
+let fs = require('fs');
+
+function readTodo(request, response) {
+  fs.readFile('./todoList.json', 'utf8', (error, todoList) => {
+    const parsedList = JSON.parse(todoList);
+    if (error) {
+      if (error.code === 'EN0ENT') {
+        console.log('No data found');
+      } else {
+        console.error(error);
+        response.status(500).send('some problem.');
+      }
+    } else if (request.params.id < parsedList.length) {
+      const todoItem = parsedList[++request.params.id]['text'];
```

Comment:
You don't want to change the ID, so don't use `++` or `--`. Also you want to change from 1-based to 0-based order, so can use `request.params.id-1` instead of `++request.params.id`
Also you may need to check zero or negative ID, and you need to accept the last element

        } else if (0 < request.params.id <= parsedList.length) {